src/tests.rs

```diff
@@ -183,3 +183,103 @@ fn basic_struct_with_phantoms() {
 
     assert_type!(SomeStruct<bool>, struct_bool_type_info);
 }
+
+#[test]
+fn basic_c_like_enum() {
+    #[allow(unused)]
+    enum BasicCStyleEnum {
+        A,
+        B,
+    }
+
+    impl TypeInfo for BasicCStyleEnum {
+        type Identity = Self;
+
+        fn type_info() -> Type {
+            Type::builder()
+                .path(Path::new("BasicCStyleEnum", module_path!()))
+                .variant(Variants::fieldless().variant("A", 0).variant("B", 1))
+        }
+    }
+
+    let ty = Type::builder()
+        .path(
+            Path::from_segments(vec!["scale_info", "tests", "BasicCStyleEnum"]).unwrap(),
+        )
+        .variant(Variants::fieldless().variant("A", 0).variant("B", 1));
+
+    assert_type!(BasicCStyleEnum, ty);
+}
+
+#[test]
+fn c_like_enum_with_discriminant_override() {
+    #[allow(unused)]
+    enum CStyleEnum {
+        A = 3,
+        B = 0,
+    }
+
+    impl TypeInfo for CStyleEnum {
+        type Identity = Self;
+
+        fn type_info() -> Type {
+            Type::builder()
+                .path(Path::new("CStyleEnum", module_path!()))
+                .variant(Variants::fieldless().variant("A", 3).variant("B", 0))
+        }
+    }
+
+    let ty = Type::builder()
+        .path(Path::from_segments(vec!["scale_info", "tests", "CStyleEnum"]).unwrap())
+        .variant(Variants::fieldless().variant("A", 3).variant("B", 0));
+
+    assert_type!(CStyleEnum, ty);
+}
+
+#[test]
+fn basic_enum() {
+    #[allow(unused)]
+    enum RustEnum {
+        A(bool),
+        B { b: u8 },
+        C(u16, u32),
+        D,
+    }
+    impl TypeInfo for RustEnum {
+        type Identity = Self;
+
+        fn type_info() -> Type {
+            Type::builder()
+                .path(Path::new("RustEnum", module_path!()))
+                .variant(
+                    Variants::with_fields()
+                        .variant("A", Fields::unnamed().field_of::<bool>("bool"))
+                        .variant("B", Fields::named().field_of::<u8>("b", "bool"))
+                        .variant(
+                            "C",
+                            Fields::unnamed()
+                                .field_of::<u16>("u16")
+                                .field_of::<u32>("u32"),
+                        )
+                        .variant_unit("D"),
+                )
+        }
+    }
+
+    let ty = Type::builder()
+        .path(Path::new("RustEnum", module_path!()))
+        .variant(
+            Variants::with_fields()
+                .variant("A", Fields::unnamed().field_of::<bool>("bool"))
+                .variant("B", Fields::named().field_of::<u8>("b", "bool"))
+                .variant(
+                    "C",
+                    Fields::unnamed()
+                        .field_of::<u16>("u16")
+                        .field_of::<u32>("u32"),
+                )
+                .variant_unit("D"),
+        );
+
+    assert_type!(RustEnum, ty);
+}
```

Comment:
These tests don't add that much value, just that `type_info` returns what it returns. Perhaps good as examples of how to manually construct type info for enums, but that would be better as doc tests.

Reply:
Fair point, I added them indeed while tinkering with the api. I'll move some to doc tests.

Reply:
The doc tests already covers this so removed them instead.